pvlib/irradiance.py

```diff
@@ -160,6 +160,12 @@ def aoi_projection(surface_tilt, surface_azimuth, solar_zenith, solar_azimuth):
 
     Input all angles in degrees.
 
+    .. warning::
```

Comment:
I would not be opposed to retiring the `aoi_projection` function. `aoi` could return cos(aoi) also, if that's of interest.

Reply:
+1 on retiring `aoi_projection`, but let the few users that need `cos(aoi)` call it themselves.

Reply:
It's not a big deal, but several of the sky diffuse transposition models use `aoi_projection()`, so that deprecation will need some other minor changes too.

Reply:
-1 on retiring `aoi_projection` because I use it.

Reply:
thanks @adriesse. I suggest we keep it and clip the output. Perhaps we should document the formula used by each function.